lib/inspec/resources/interfaces.rb

```diff
@@ -0,0 +1,118 @@
+require "inspec/utils/filter"
+require "inspec/resources/command"
+
+module Inspec::Resources
+  class Interfaces < Inspec.resource(1)
+    name "interfaces"
+    supports platform: "unix"
+    supports platform: "windows"
+    desc "Use the interfaces InSpec audit resource to test properties for multiple network interfaces installed on the system"
+    example <<~EXAMPLE
+      describe interfaces do
+        its('names') { should include 'eth0' }
+      end
+    EXAMPLE
+
+    attr_reader :iface_data
+
+    def to_s
+      "Interfaces"
+    end
+
+    filter = FilterTable.create
+    filter.register_column(:names, field: "name")
+      .install_filter_methods_on_resource(self, :scan_interfaces)
+
+    def ipv4_address
+      # Loop over interface names
+      # Select those that are up and have an ipv4 address
+      interfaces = names.map { |n| inspec.interface(n) }.select do |i|
+        i.ipv4_address? && i.up?
+      end
+
+      all_addrs = interfaces.map(&:ipv4_addresses).flatten
+
+      # Look for progressively "better" IP addresses
+
+      # Reject anything that looks loopback-ish
+      non_loopback_addrs = all_addrs.reject { |a| a =~ /^127/ }
+      return all_addrs.first if non_loopback_addrs.empty?
+
+      # OK, we have something that isn't loopbackish.
+      # Try to filter out management networks.
+      non_management_addrs = non_loopback_addrs.reject { |a| a =~ /^10\./ }
```

Comment:
I think `10.0.0.0/8` is just a private IP address range alongside `172.16.0.0/12` and `192.168.0.0/16` 

Maybe 10.x is used for management layers but I've also seen it used for workstation IP addresses.

I think these 3 ranges should be treated equally (some other IP address that _might_ be a public IP is preferable). 

https://en.wikipedia.org/wiki/Private_network#Private_IPv4_addresses

Reply:
OK, I've reworked this section to clarify what is happening and remove the variable names which seem to be objectionable.  I've added 172.16/12 as well.